File: apps/api/src/lib/password.ts
import bcrypt from 'bcryptjs';
import { config } from '../config.js';

export async function hashPassword(plain: string): Promise<string> {
  return bcrypt.hash(plain, config.bcryptRounds);
}

export async function verifyPassword(plain: string, hash: string): Promise<boolean> {
  return bcrypt.compare(plain, hash);
}

/** Minimal password policy. Returns an error message or null if valid. */
export function validatePasswordStrength(plain: string): string | null {
  if (plain.length < 10) return 'Password must be at least 10 characters';
  return null;
}

File: apps/api_py/lib/password.py
import re
import bcrypt
import config as cfg


def hash_password(plaintext: str) -> str:
    rounds = cfg.BCRYPT_ROUNDS
    salt = bcrypt.gensalt(rounds=rounds)
    return bcrypt.hashpw(plaintext.encode("utf-8"), salt).decode("utf-8")


def verify_password(plaintext: str, hashed: str) -> bool:
    try:
        return bcrypt.checkpw(plaintext.encode("utf-8"), hashed.encode("utf-8"))
    except Exception:
        return False


def validate_password_strength(password: str) -> str | None:
    """Returns an error message or None if valid."""
    if len(password) < 10:
        return "Password must be at least 10 characters"
    if not re.search(r"[A-Z]", password):
        return "Password must contain at least one uppercase letter"
    if not re.search(r"[0-9]", password):
        return "Password must contain at least one digit"
    return None
